test/unittest/layers/layers_golden_tests.cpp

```diff
@@ -227,6 +227,11 @@ TEST_P(LayerGoldenTest, run) {
   bool skip_calc_grad = shouldSkipCalcGrad();
   bool skip_calc_deriv = shouldSkipCalcDeriv();
 
+  for (int i = 0; i < 4; ++i) {
+    /// warm layer multiple times
+    layer->forwarding(rc, !shouldForwardWithInferenceMode());
+  }
+
   layer->forwarding(rc, !shouldForwardWithInferenceMode());
```

Comment:
this forwards 5 times in total but the reference code is only forwarding 4 times. I wonder how the unittests are passing?

Reply:
test/input_gen/recorder.py:407 has extra one forward more, corresponeding with current structure

Reply:
I am not sure but test/input_gen/recorder.py:407 just seems to have a loop of 4, but in cpp code, there is a loop of 4 + an extra forward outside the loop. Please check.

Reply:
Sorry, it's 407 in the old PR but it's 411. in the new PR

https://github.com/nnstreamer/nntrainer/pull/1589/files/ffc62cc06223bc1beed96b86deeca1e2a084893b..209f02c9972c8f72cfeed1c6804dbac9cfb1ae47#diff-5cd4cf5b490627ad8cb536f2ceaf55b7361475614f8b33873a5879dc7d0cb20dR411

Reply:
aah got it. It might be good to add some comments related to this in both py and cpp code.